torchtitan/components/checkpoint.py

```diff
@@ -231,16 +232,41 @@ def __init__(
         lr_schedulers: LRSchedulersContainer,
         states: Dict[str, Any],
         job_config: JobConfig,
+        ft_manager: Optional["ft.Manager"] = None,
     ) -> None:
         ckpt_config = job_config.checkpoint
         self.enable_checkpoint = ckpt_config.enable_checkpoint
+        self.ft_manager = ft_manager
+
+        if self.ft_manager:
+            optimizers.init_cache_state_dict()
```

Comment:
Do we only need to call this once or once per step? Tensors never change?

Reply:
We also call this when `optimizers.load_state_dict()` is called.